make configure.ac POSIX compliant

https://github.com/LinuxCNC/linuxcnc/blob/033f83d20a069754157fc2d0765d9deb9cdc4599/src/configure.ac#L98

Must be replaced to make it [POSIX](https://en.wikipedia.org/wiki/POSIX) compliant, as it follows: 
` 
CFLAGS="${CFLAGS} `pkg-config libtirpc --cflags`" 
`

I'm using Dash instead of Bash in Gentoo Linux, it doesn't recognize "+=" instruction. With this change it will comply with all other kind of shells used by different Linux distributions.

> The ${:+} notation is POSIX compliant, as far as I know.  I know it work with dash.

If it is the case, Why the executable generated by this piece of code fail when run with Dash but not with Bash?

configure.ac :
`
#!/bin/sh
#-*- Autoconf -*-
#Process this file with autoconf to produce a configure script.

AC_PREREQ([2.71])
AC_INIT([FULL-PACKAGE-NAME], [VERSION], [BUG-REPORT-ADDRESS])

#Checks for header files.
#check for rpc/rpc.h
AC_CHECK_HEADER([rpc/rpc.h],[], [
  if ! test -z "$PKG_CONFIG libtirpc"; then


    LIBTIRPC_CFLAGS=\`pkg-config libtirpc --cflags\`
    AC_SUBST([LIBTIRPC_CFLAGS])
    AC_MSG_WARN([$LIBTIRPC_CFLAGS])

    LIBTIRPC_LIBS=\`pkg-config libtirpc --libs\`
    AC_SUBST([LIBTIRPC_LIBS])
    AC_MSG_WARN([$LIBTIRPC_LIBS])

    AC_MSG_WARN([$CFLAGS])
    #CFLAGS="${CFLAGS} \`pkg-config libtirpc --cflags\`"
    CFLAGS+=" \`pkg-config libtirpc --cflags\` "
    AC_MSG_WARN([$CFLAGS])
  fi

  #Update path
  AC_CHECK_LIB([tirpc], [get_myaddress], [], [
    AC_MSG_WARN([papas])
    AC_MSG_ERROR([Didn't find tirpc])
  ])

  #Clear the cached check
  $as_unset ac_cv_header_rpc_rpc_h
  AC_CHECK_HEADER([rpc/rpc.h],[],[
    AC_MSG_ERROR([glibc rpc.h missing, please install libtirpc])
  ])
])

AC_OUTPUT
`

$ autoconf
$ dash configure
output:

`
checking for rpc/rpc.h... no
configure: WARNING: -I/usr/include/tirpc 
configure: WARNING: -ltirpc
configure: WARNING: -g -O2
configure: 3273: CFLAGS+= -I/usr/include/tirpc : not found
configure: WARNING: -g -O2
checking for get_myaddress in -ltirpc... yes
checking for rpc/rpc.h... no
configure: error: glibc rpc.h missing, please install libtirpc
`

I'm using Dash-0.5.12

--- a/src/configure.ac
+++ b/src/configure.ac
@@ -97,7 +97,7 @@ AC_CHECK_HEADER([rpc/rpc.h],[], [
     LIBTIRPC_LIBS=`pkg-config libtirpc --libs`
     AC_SUBST([LIBTIRPC_LIBS])
 
-    CFLAGS+=" `pkg-config libtirpc --cflags` "
+    CFLAGS="${CFLAGS:+$CFLAGS }$(pkg-config libtirpc --cflags)"
   fi
 
   # Update path
@@ -411,8 +411,8 @@ fi
 # Set flags for C and C++ if supported
 for commonflag in -Wno-stringop-truncation ; do
   if echo "int main() { return 0;}" | $CC -Werror $commonflag -E - > /dev/null; then
-    CFLAGS="$CFLAGS $flag"
-    CXXFLAGS="$CXXFLAGS $flag"
+    CFLAGS="${CFLAGS:+$CFLAGS }$flag"
+    CXXFLAGS="${CXXFLAGS:+$CXXFLAGS }$flag"
   fi
 done
 

